packages/browser-wallet/src/popup/pages/Setup/ChooseNetwork.tsx

```diff
@@ -1,23 +1,29 @@
 import PageHeader from '@popup/shared/PageHeader';
 import React from 'react';
-import { useNavigate } from 'react-router-dom';
+import { useLocation, useNavigate } from 'react-router-dom';
 import Button from '@popup/shared/Button';
 import { absoluteRoutes } from '@popup/constants/routes';
-import { useSetAtom, useAtomValue } from 'jotai';
+import { useSetAtom } from 'jotai';
 import { useTranslation } from 'react-i18next';
-import { isRecoveringAtom } from '@popup/state';
-import { networkConfigurationAtom } from '@popup/store/settings';
+import { hasBeenOnBoardedAtom, networkConfigurationAtom, sessionPasscodeAtom } from '@popup/store/settings';
 import { setupRoutes } from './routes';
 // TODO Remove stagenet
 import { mainnet, testnet, stagenet } from '../NetworkSettings/NetworkSettings';
+import { usePasscodeInSetup } from './passcode-helper';
 
 export function ChooseNetwork() {
     const navigate = useNavigate();
+    const location = useLocation();
+    const isRecovering = location.pathname.endsWith('recovering');
     const { t } = useTranslation('setup');
     const setNetworkConfiguration = useSetAtom(networkConfigurationAtom);
-    const isRecovering = useAtomValue(isRecoveringAtom);
+    const setHasBeenOnboarded = useSetAtom(hasBeenOnBoardedAtom);
+    const passcode = usePasscodeInSetup();
+    const setPasscodeInSession = useSetAtom(sessionPasscodeAtom);
 
     const goToNext = () => {
```

Comment:
This should probably go to `absoluteRoutes.home.identities.add.path` when not recovering.

Reply:
Are you sure? I would think we (in the future) would just add a button or something to the identities page for when you have no identities. I am not updating it now.